### PR TITLE
[8.4] MOD-13323: revert cursor sub-commands in 8.4

### PR DESCRIPTION
**Pull request by oshadmi**
## Describe the changes in the pull request

A clear and concise description of what the PR is solving, including:
1. Current: The current state briefly
2. Change: What is the change
3. Outcome: Adding the outcome

#### Which additional issues this PR fixes
1. MOD-...
2. #...

#### Main objects this PR modified
1. ...

#### Mark if applicable

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

---

> [!NOTE]
> Unifies `FT.CURSOR` handling and updates registration, docs, and tests.
> 
> - Replace `RSCursorRead/Del/Profile/GC` with a single `RSCursorCommand` that internally dispatches `READ|PROFILE|DEL|GC`; wire through both OSS and cluster paths
> - Register `FT.CURSOR` as one command (`SetFtCursorInfo`) and update `command_info`/`COMMAND INFO` metadata (includes `request_policy:special`)
> - Update `commands.json`: collapse `FT.CURSOR READ/DEL` into a single `FT.CURSOR` entry
> - Adjust distributed execution to call the unified handler; remove subcommand registration logic/macros
> - Tests updated to use unified command, verify tips, adjust error messages and `COMMANDSTATS` keys; remove GC edge-case test relying on old behavior
> - Minor behavior tweaks: consolidated GC return value, standardized "Unknown subcommand"/bad ID replies
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 11f0a84c68916169f6af362825f4195385602cc1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/aggregate/aggregate_exec.c [hunk 1]
```diff
@@ -1385,144 +1383,116 @@ static void cursorRead_ctx(CursorReadCtx *cr_ctx) {
 
 /**
  * FT.CURSOR READ {index} {CID} {COUNT} [MAXIDLE]
- */
-int RSCursorReadCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  if (argc < 4) {
-    return RedisModule_WrongArity(ctx);
-  }
-
```
[the remark this answers is not in the document]
Comment (Itzikvaknin) [reply src/aggregate/aggregate_exec.c:1393]: This is the same as it was before the subcommands PR, and as subcommands it is easier to address this issue
so Ill leave it outside of this PR

File: src/aggregate/aggregate_exec.c [hunk 2]
```diff
@@ -1385,144 +1383,116 @@ static void cursorRead_ctx(CursorReadCtx *cr_ctx) {
...
+    int rc = Cursors_Purge(GetGlobalCursor(cid), cid);
+    if (rc != REDISMODULE_OK) {
+      RedisModule_Reply_Error(reply, "Cursor does not exist");
+    } else {
+      RedisModule_Reply_SimpleString(reply, "OK");
+    }
+  } else if (strcasecmp(cmd, "GC") == 0) {
+    int rc = Cursors_CollectIdle(&g_CursorsList);
+    rc += Cursors_CollectIdle(&g_CursorsListCoord);
+    RedisModule_Reply_LongLong(reply, rc);
```
[the remark this answers is not in the document]
Comment (Itzikvaknin) [reply src/aggregate/aggregate_exec.c]: No it didn't

**Comment by oshadmi** [comment]
@Itzikvaknin LGTM

File: src/command_info/command_info.c [hunk 3]
```diff
@@ -2178,42 +2178,11 @@ int SetFtProfileInfo(RedisModuleCommand *cmd) {
...
-        .summary = "Specifies the name of the index. The index must be created using `FT.CREATE`.",
-        .type = REDISMODULE_ARG_TYPE_STRING,
-      },
-      {
-        .name = "cursor_id",
-        .type = REDISMODULE_ARG_TYPE_INTEGER,
-      },
-      {
-        .name = "read size",
-        .token = "COUNT",
```
[the remark this answers is not in the document]
Comment (Itzikvaknin) [reply src/command_info/command_info.c:2199]: Known issue, out of the scope of this PR
